plugins/outputs/prometheus_client/prometheus_client.go

```diff
@@ -6,31 +6,40 @@ import (
 	"net/http"
 	"regexp"
 	"sync"
+	"time"
 
 	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
 	"github.com/influxdata/telegraf/plugins/outputs"
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 var invalidNameCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
 
+type MetricWithExpiration struct {
+	Metric     prometheus.Metric
+	Expiration time.Time
+}
+
 type PrometheusClient struct {
-	Listen string
+	Listen             string
+	ExpirationInterval internal.Duration `toml:"expiration_interval"`
 
-	metrics     map[string]prometheus.Metric
-	lastMetrics map[string]prometheus.Metric
+	metrics map[string]*MetricWithExpiration
 
 	sync.Mutex
 }
 
 var sampleConfig = `
   ## Address to listen on
   # listen = ":9126"
+
+  ## Interval to expire metrics and not deliver to prometheus, 0 == no expiration
+  # expiration_interval = 0
```

Comment:
I would say this should default to `"60s"`, since the current behavior of the plugin is to expire metrics.